components/aws/sagemaker/tests/integration_tests/component_tests/test_hpo_component.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+import os
+import json
+import utils
+
+from utils import kfp_client_utils
+from utils import minio_utils
+from utils import sagemaker_utils
+
+
+@pytest.mark.parametrize("test_file_dir", ["resources/config/kmeans-mnist-hpo"])
+def test_hyperparameter_tuning(
+    kfp_client, experiment_id, region, sagemaker_client, test_file_dir
+):
+
+    download_dir = utils.mkdir(os.path.join(test_file_dir + "/generated"))
+    test_params = utils.load_params(
+        utils.replace_placeholders(
+            os.path.join(test_file_dir, "config.yaml"),
+            os.path.join(download_dir, "config.yaml"),
+        )
+    )
+
+    test_params["Arguments"]["channels"] = json.dumps(
+        test_params["Arguments"]["channels"]
+    )
+    test_params["Arguments"]["static_parameters"] = json.dumps(
+        test_params["Arguments"]["static_parameters"]
+    )
+    test_params["Arguments"]["integer_parameters"] = json.dumps(
+        test_params["Arguments"]["integer_parameters"]
+    )
+    test_params["Arguments"]["categorical_parameters"] = json.dumps(
+        test_params["Arguments"]["categorical_parameters"]
+    )
+
+    run_id, status, workflow_json = kfp_client_utils.compile_run_monitor_pipeline(
```

Comment:
nit: These variables are not being used in most cases, there is no need to assign them.